api/nodes/serializers.py

```diff
@@ -1497,6 +1497,32 @@ class DraftRegistrationSerializer(JSONAPISerializer):
     def get_absolute_url(self, obj):
         return obj.absolute_url
 
+    def update_metadata(self, draft, metadata, reviewer, required_fields=False):
+        try:
+            # Required fields are only required when creating the actual registration, not updating the draft.
+            draft.validate_metadata(metadata=metadata, reviewer=reviewer, required_fields=required_fields)
+        except ValidationError as e:
+            raise exceptions.ValidationError(e.message)
+        draft.update_metadata(metadata)
+        draft.save()
+
+    def update_registration_responses(self, draft, registration_responses, required_fields=False):
+        # New workflow - at some point `registration_metadata` will be deprecated, but for now,
+        # we support data coming in on either field, registration_metadata (expanded) or registration_responses (flat)
+        try:
+            draft.validate_registration_responses(registration_responses=registration_responses, required_fields=required_fields)
+        except ValidationError as e:
+            raise exceptions.ValidationError(e.message)
+        draft.update_registration_responses(registration_responses)
+        draft.save()
+
+    def enforce_metadata_or_registration_responses(self, metadata=None, registration_responses=None):
+        if metadata and registration_responses:
+            raise exceptions.ValidationError(
+                'You cannot include both `registration_metadata` and `registration_responses` in your request. Please use' +
+                ' `registration_responses` as `registration_metadata` will be deprecated in the future.',
+            )
```

Comment:
While we support both workflows, writing to `registration_metadata` or `registration_responses` will build the corresponding data and write to the other field. So we don't want to be writing to both!